src/main/java/com/teragrep/lsh_01/pool/ManagedRelpConnection.java

```diff
@@ -26,19 +26,19 @@
 import java.io.IOException;
 import java.util.concurrent.TimeoutException;
 
-public class ManagedRelpConnection {
+public class ManagedRelpConnection implements IManagedRelpConnection {
 
     private static final Logger LOGGER = LogManager.getLogger(ManagedRelpConnection.class);
     private final IRelpConnection relpConnection;
+    private boolean connected;
 
     public ManagedRelpConnection(IRelpConnection relpConnection) {
         this.relpConnection = relpConnection;
+        this.connected = false;
     }
 
-    void connect() {
-        boolean notConnected = true;
-        while (notConnected) {
-            boolean connected = false;
+    private void connect() {
+        while (!connected) {
```

Comment:
please do not use class member field here, but a local variable, update the member field only after the connection has taken place. this is because 'connected' will not follow the actual status of the network connection. the actual status can not be predetermined and can change at any point in time. it is totally random in that sense. the only reason to connect before sending at least once is to mitigate https://github.com/teragrep/rlp_01/issues/63

Reply:
Changed connected to hasConnected (opposed to isConnected, which was its role even though it had bad naming). So now it only follows if a connection has been made at least once so it can mitigate the rlp_01 bug, it doesn't follow the actual status.